lib/model/port.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * port.js
+ *
+ * Definition for the Port class. Ports are the points on a Node where
+ * connections attach to or from the node. They can be either inputs
+ * or outputs.
+ */
+
+/**
+ * Constructor function.
+ *
+ * @param id - The Port's unique identifier.
+ * @param type - The type of port. Valid options include "input" or "output".
+ */
+var Port = function(id,type) {
+	this.id = id || 0;
+	this.x = 0;

```

Comment:
It gets converted, that x=0 in the constructor is just initializing the attribute in the object, which I think is the proper way to do that in JS. If you notice the function cardinalToCoordinate converts the fromat from the layout spec (N 50 for the example above) into what I specified.